test/e2e/apimachinery/aggregator.go

```diff
@@ -159,36 +159,24 @@ func setupSampleAPIServerCert(namespaceName, serviceName string) *aggregatorCont
 
 // A basic test if the sample-apiserver code from 1.7 and compiled against 1.7
 // will work on the current Aggregator/API-Server.
-func TestSampleAPIServer(f *framework.Framework, image, namespaceName string) {
+func TestSampleAPIServer(f *framework.Framework, image string) {
 	By("Registering the sample API server.")
 	cleanTest(f, true)
 	client := f.ClientSet
 	restClient := client.Discovery().RESTClient()
 	iclient := f.InternalClientset
 	aggrclient := f.AggregatorClient
 
-	context := setupSampleAPIServerCert(namespaceName, "sample-api")
-	ns := f.Namespace.Name
+	namespace := f.Namespace.Name
+	context := setupSampleAPIServerCert(namespace, "sample-api")
 	if framework.ProviderIs("gke") {
 		// kubectl create clusterrolebinding user-cluster-admin-binding --clusterrole=cluster-admin --user=user@domain.com
 		authenticated := rbacv1beta1.Subject{Kind: rbacv1beta1.GroupKind, Name: user.AllAuthenticated}
-		framework.BindClusterRole(client.RbacV1beta1(), "cluster-admin", ns, authenticated)
+		framework.BindClusterRole(client.RbacV1beta1(), "cluster-admin", namespace, authenticated)
 	}
 
 	// kubectl create -f namespace.yaml
-	var namespace string
-	err := wait.Poll(100*time.Millisecond, 30*time.Second, func() (bool, error) {
-		got, err := client.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}})
-		if err != nil {
-			if strings.HasPrefix(err.Error(), "object is being deleted:") {
-				return false, nil
-			}
-			return false, err
-		}
-		namespace = got.Name
-		return true, nil
-	})
-	framework.ExpectNoError(err, "creating namespace %q", namespaceName)
+	// NOTE: use the namespace in test framework (normally user will create a namespace)
```

Comment:
Please add comment to the effect of aggregated apis should generally be set up in there own namespace. As the test framework is setting up a new namespace, we are just using that.